src/service/action-impl.js

```diff
@@ -726,6 +748,23 @@ class ParserTokenizer {
       return {type: TokenType.LITERAL, value, index: newIndex};
     }
 
+    // Object literal.
+    if (c == OBJECT_SET[0]) { // '{'
+      let end = -1;
+      for (let i = newIndex + 1; i < this.str_.length; i++) {
```

Comment:
What I mean is that `{ a: {b: 1 } }` will cause an error, since we'll start processing this object expression at `a`, then close it after `b`. But now we have an extra `}` character. You'll need to keep track of the number of open `{`, and only close once we've reached the same number of `}`.